packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -110,37 +146,57 @@ export function cloudflare<
 								type: 'ESModule',
 								path: runnerPath,
 							},
+							{
+								// we declare the workerd-custom-import as a CommonJS module, thanks to this
+								// require is made available in the module and we are able to handle cjs imports, etc...
+								type: 'CommonJS',
+								path: fileURLToPath(
+									new URL('workerd-custom-import.cjs', import.meta.url),
+								),
+							},
 						],
 						serviceBindings: {
 							...workerOptions.serviceBindings,
 							__VITE_FETCH_MODULE__: async (request) => {
-								const args = (await request.json()) as [
-									string,
-									string,
-									FetchFunctionOptions,
-								];
+								const [moduleId, imported, options] =
+									(await request.json()) as [
+										string,
+										string,
+										FetchFunctionOptions,
+									];
+
+								if (moduleId.startsWith('cloudflare:')) {
+									const result = {
+										externalize: moduleId,
+										type: 'builtin',
+									} satisfies vite.FetchResult;
```

Comment:
Can this externalization not be handled by configuring Vite rather than hard coding here?

Reply:
mh.... that's a good point! 🤔

I think we can, I'll have a look 🙂👍

Reply:
We also need to add them (`cloudflare:workers`, `cloudflare:sockets` etc.) to `resolve.external` on the environment so that they will be externalised in the build but I'm not sure that will replace this behaviour in dev. Worth trying but if not let's keep build only stuff for a separate PR.

Reply:
it seems like in vite this whole builtin handling is done in a pretty hardcoded manner: 
https://github.com/vitejs/vite/blob/889bfc0ada6d6cd356bb7a92efdce96298f82fef/packages/vite/src/node/utils.ts#L95-L118

🤔

Reply:
I am leaning into thinking that this needs to be hardcoded to a certain degree... maybe it is actually valid to put it as the last check in the method, after everything else has failed...

meaning that we resolve a `cloudflare:*` import as a builtin one only after everything else has failed, meaning that if the env consumer wants to polyfill the import or whatever they'd like we'd still enable that 🤔

Reply:
Good find. Would it make sense to get `cloudflare:` added to this list? It looks like the `isBuiltIn` function gets called by `fetchModule` - https://github.com/vitejs/vite/blob/889bfc0ada6d6cd356bb7a92efdce96298f82fef/packages/vite/src/node/ssr/fetchModule.ts#L32. I guess this still wouldn't work if `cloudflare:` is imported inside `node_modules`.

Reply:
> Would it make sense to get `cloudflare:` added to this list?

I've asked this in the discord channel, I feel like that might not be the right approach, but it's worth double checking I guess 🙂

Reply:
> I guess this still wouldn't work if `cloudflare:` is imported inside `node_modules`.

No this would not get triggered by code from `node_modules`, but it would still work totally fine since it runs outside of the vite module resolution

To be extra sure I've added tests for this: https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/15/commits/bddf610f7a509823dd134e988f0523ca7c3eee44

Reply:
> I've asked this in the discord channel, I feel like that might not be the right approach, but it's worth double checking I guess 🙂

[They told me](https://discord.com/channels/804011606160703521/831456449632534538/1293625957994987551) that `cloudflare:` should not be included there, and that node, deno and bun are because apparently those are all platforms that can run a vite dev server 😮